jupyterhub/spawner.py

```diff
@@ -1308,12 +1358,17 @@ def get_env(self):
         we return the exit code of the process if we have access to it, or 0 otherwise.
         """
         # if we started the process, poll with Popen
-        if self.proc is not None:
-            status = self.proc.poll()
-            if status is not None:
-                # clear state if the process is done
+        if isinstance(self.proc_state, Queue_class):
```

Comment:
This isinstance check appears irrelevant, since proc_state is always either None or this Queue, right?